src/requestforattestation/RequestForAttestation.spec.ts

```diff
@@ -540,58 +479,43 @@ describe('RequestForAttestation', () => {
         []
       )),
     } as IRequestForAttestation
-    builtRequestMalformedCtypeHash.cTypeHash = {
-      hash: [
-        builtRequestMalformedCtypeHash.cTypeHash.hash.slice(0, 15),
-        (
-          (parseInt(
-            builtRequestMalformedCtypeHash.cTypeHash.hash.charAt(15),
-            16
-          ) +
-            1) %
-          16
-        ).toString(16),
-        builtRequestMalformedCtypeHash.cTypeHash.hash.slice(16),
-      ].join(''),
-      nonce: builtRequestMalformedCtypeHash.cTypeHash.nonce,
-    }
-    builtRequestMalformedCtypeHash.rootHash = RequestForAttestation[
-      'calculateRootHash'
-    ](
-      builtRequestMalformedCtypeHash.claimOwner,
-      builtRequestMalformedCtypeHash.cTypeHash,
-      builtRequestMalformedCtypeHash.claimHashTree,
-      builtRequestMalformedCtypeHash.legitimations,
-      builtRequestMalformedCtypeHash.delegationId
+    Object.entries(builtRequestMalformedHashes.claimNonceMap).forEach(
+      ([hash, nonce]) => {
+        const scrambledHash = [
+          hash.slice(0, 15),
+          ((parseInt(hash.charAt(15), 16) + 1) % 16).toString(16),
+          hash.slice(16),
+        ].join('')
+        builtRequestMalformedHashes.claimNonceMap[scrambledHash] = nonce
+        delete builtRequestMalformedHashes.claimNonceMap[hash]
+      }
     )
+    builtRequestMalformedHashes.rootHash = RequestForAttestation[
+      'calculateRootHash'
+    ](builtRequestMalformedHashes)
     expect(() =>
       RequestForAttestationUtils.errorCheck(builtRequestNoLegitimations)
     ).toThrowError(ERROR_LEGITIMATIONS_NOT_PROVIDED())
     expect(() =>
       RequestForAttestationUtils.errorCheck(builtRequestMalformedRootHash)
     ).toThrowError(ERROR_ROOT_HASH_UNVERIFIABLE())
-    expect(() =>
-      RequestForAttestationUtils.errorCheck(builtRequestMalformedClaimOwner)
-    ).toThrowError(
-      ERROR_NONCE_HASH_INVALID(
-        builtRequestMalformedClaimOwner.claimOwner,
-        'Claim owner'
-      )
-    )
     expect(() =>
       RequestForAttestationUtils.errorCheck(builtRequestIncompleteClaimHashTree)
-    ).toThrowError(ERROR_CLAIM_HASHTREE_MALFORMED())
+    )
+      .toThrowError
+      // TODO: fix error type
+      // ERROR_CLAIM_HASHTREE_MALFORMED()
+      ()
     expect(() =>
       RequestForAttestationUtils.errorCheck(builtRequestMalformedSignature)
     ).toThrowError(ERROR_SIGNATURE_UNVERIFIABLE())
     expect(() =>
-      RequestForAttestationUtils.errorCheck(builtRequestMalformedCtypeHash)
-    ).toThrowError(
-      ERROR_NONCE_HASH_INVALID(
-        builtRequestMalformedCtypeHash.cTypeHash,
-        'CType'
-      )
+      RequestForAttestationUtils.errorCheck(builtRequestMalformedHashes)
     )
+      .toThrowError
+      // TODO: fix error type
+      // ERROR_NONCE_HASH_INVALID()
```

Comment:
What needs to be fixed here?

Reply:
I haven't figured out error reporting for this. It seems like the old sdk errors don't fully apply any more. We had `ERROR_CLAIM_HASHTREE_MALFORMED`, `ERROR_NONCE_HASH_INVALID`, `ERROR_NONCE_HASH_MALFORMED`, `ERROR_ROOT_HASH_UNVERIFIABLE`, `ERROR_CLAIM_UNVERIFIABLE`
Now there's either an unsubstantiated statement for which there is no matching digest in the nonce map, or there is a digest and nonce for a statement, but it they do not combine into a hash that's included on the signed credential. I probably have to create new error types for this. Maybe something like NO_PROOF_FOR_STATEMENT and INVALID_PROOF_FOR_STATEMENT ?
Also the way verification currently works I'm not even sure I can properly differentiate between the two

Reply:
The high level errors should be handled in another ticket. Please make one.
Please provide the low level errors as part of this PR and remove the now unthrown low level errors.